packets1/willmsgreq.go

```diff
@@ -24,7 +24,7 @@ func (p *WillMsgReq) Pack() ([]byte, error) {
 }
 
 func (p *WillMsgReq) Unpack(buf []byte) error {
-	if len(buf) <= int(willMsgReqVarPartLength) {
+	if len(buf) != int(willMsgReqVarPartLength) {
```

Comment:
I grepped `len(buf) <=` and there are couple of instances similar to this one in the code. Could you check them and fix them if they are instances of the same problem?

Reply:
I see. This looks like the only instance of the bug then.